TripalCultivate/TripalCultivate-Phenotypes
# G4.91 Remove all deprecations

**Issue TripalCultivate/TripalCultivate-Phenotypes#91**
This is also the last task needed to complete Issue TripalCultivate/TripalCultivate-Phenotypes#82!

CLOSES TripalCultivate/TripalCultivate-Phenotypes#82 
But is just progress on TripalCultivate/TripalCultivate-Phenotypes#91, the next PR will address coding standards cleanup.

## Motivation
To remove all mention of scope and outdated methods, such as the `getValidatorIdWithScope` method. These will be removed from all validators, importers, the annotation and plugin interface.

Any changes made in this PR should work with the existing infrastructure of the module. The trait importer should be able to be used through the UI.

## What does this PR do?
 - [x] Ensures the validation UI renders the messages returned by processValidationMessages().
     - [x] formValidate now skips empty lines before validating the raw row, since all of our example files were failing due to VS code automatically adding empty rows at the end of each file
     - [x] Added an additional scenario to formValidateTest to ensure line numbers are reported correctly when empty rows are involved
     - [x] Modified "file_types" in the TraitImporter annotation to be only `tsv` instead of both `tsv` and `txt`, as `splitRowIntoColumns()` doesn't accomodate mime-types with more than one possible file extension. See Issue TripalCultivate/TripalCultivate#46 
 - [x] Removes the `$old_headers` property and resolve any of its dependencies in the TraitsImporter class to use the new `$headers` property
 - [x] Removes deprecated properties from the `ValidatorBase` class: `$project, $genus, $file_id, $headers, $skip`
 - [x] Removes method `loadAssets()` from `ValidatorInterface` and `ValidatorBase`
 - [x] Removes method `getValidatorScope()` from `ValidatorInterface` and `ValidatorBase`
 - [x] Removes method `validate()` from `ValidatorInterface` and `ValidatorBase`
 - [x] Removes method `getValidatorIdWithScope()` from `ValidatorManager`
 - [x] Removes the `$validator_scope` property from the `Validator` class
 - [x] Removes an array of counts of validation status for each row of a data file that were added in PR TripalCultivate/TripalCultivate-Phenotypes#90. In case we later find we need this code, a patch has been made and attached to Issue TripalCultivate/TripalCultivate-Phenotypes#114 since this is the most likely place where we'll find out.

## Testing

### Automated Testing
No tests were added in this PR, but some asserts have been removed where deprecated methods were used:
- `getValidatorScope` in `testBasicValidatorGetters()`
- `validate` in `testValidatorValidateMethods()`

I lied, one scenario was added to the data provider in `TraitImporterFormValidateTest`:
- Tests file `correct_header_emptycell_after_empty_line.tsv` which is the same scenario as `correct_header_emptycell_method.tsv` but with an empty line added to line 3 (previous line 3 is bumped to line 4).

### Manual Testing

1. You can search the codebase for this PR for the following keywords:
- `deprecate` should bring up no results
- `scope` will bring up a bunch of hits in `TripalCultivatePhenoshareImporter.php` but these can be ignored as that importer will be dealt with in a separate PR. The remaining hits should all be in `*.md` files where scope has a different meaning than its use in our original design for this module.
- Try out other keywords you can think of that might reference the previous design, like old methods or property values.
- Note that `todo` is still referenced A LOT due to being in the actual code as a validation status and there are more `@todo`s that are outside the scope of this PR since they need a lot of work.

2. To test that the module is working appropriately through the interface:
   1. Clone this branch and build a docker off of it
   2. Login to your site and create a new organism at `Tripal > Content > Add Tripal Content > Organism`
   3. Once your organism was successfully created, navigate to `Tripal > Extensions > Tripal Cultivate Phenotypes Configuration ` Go to Ontology Terms and select vocabularies for every term, just ensure they aren't all the same vocab :) Click on `Save Configuration`
   4. Navigate to `Tripal > Data Loaders > Tripal Cultivate: Phenotypic Trait Importer`
   5. Make sure your genus from Step ii) is selected, and then upload a file to test. You can find examples in `trpcultivate_phenotypes/tests/src/Fixtures/TraitImporterFiles`, ***however*** you must select:
      - a TSV file, since TXT is not yet supported
      - NOT `correct_header_no_data.tsv` since there is a known bug with this case 🐞 See Issue TripalCultivate/TripalCultivate-Phenotypes#117 
   6. Check that the validation feedback makes sense for the file uploaded. If you need help figuring out what should pass and what shouldn't, search for the file you used in `TraitImporterFormValidateTest.php` to see the expected statuses of that scenario.
